Second test case using embedded Neo4j fails with "PortBindException: Address already in use"

My project uses Micronaut 2.0.0.M3/RC2 building a Neo4j-backed application. I've written two test-cases, each one tests different HTTP endpoints of the application. Both test-cases make use of [embedded Neo4J](https://micronaut-projects.github.io/micronaut-neo4j/2.0.0/guide/index.html#testing).

With every execution of test suites the test case executed as second one fails with root cause `Caused by: org.neo4j.helpers.PortBindException: Address localhost:7687 is already in use, cannot bind to it.`

To me it seems that test-cases does not shut down the embedded Neo4j database (although something like `server.stop()` or `applicationContext.stop()` is invoked explicitely), so when the second test-case tries to launch its own embedded instance, the usual Neo4j port is still in use.

### Steps to Reproduce

1. To reproduce the problem, I've added a second test case to project's already available Spock Specification file `io.micronaut.configuration.neo4j.bolt.Neo4jEmbeddedServerSpec` (branch `micronaut-2x`) like this:
```groovy
import io.micronaut.context.ApplicationContext
import io.micronaut.core.io.socket.SocketUtils
import org.neo4j.driver.Driver
import spock.lang.Specification

class Neo4jEmbeddedServerSpec extends Specification{

    static int EMBEDDED_NEO4J_TCP_PORT = SocketUtils.findAvailableTcpPort()

    void "test neo4j embedded"() {
        given:
        ApplicationContext applicationContext = ApplicationContext.run(
                'neo4j.uri':"bolt://localhost:${EMBEDDED_NEO4J_TCP_PORT}",
                'neo4j.embedded.ephemeral':true
        )

        when:
        Driver driver = applicationContext.getBean(Driver)

        then:
        driver.session().run('MATCH (n) RETURN n').size() == 0

        cleanup:
        applicationContext?.stop()
    }

    void "check embedded neo4j has been shut down by first test case"() {
        given:
        ApplicationContext applicationContext = ApplicationContext.run(
                'neo4j.uri':"bolt://localhost:${EMBEDDED_NEO4J_TCP_PORT}",
                'neo4j.embedded.ephemeral':true
        )

        when:
        Driver driver = applicationContext.getBean(Driver)

        then:
        driver.session().run('MATCH (n) RETURN n').size() == 0

        cleanup:
        applicationContext?.stop()
    }
}
```
2. Running `./gradlew clean build` reproduces my observation:
```
io.micronaut.configuration.neo4j.bolt.Neo4jEmbeddedServerSpec > check embedded neo4j has been shut down by first test case FAILED
[...]
Caused by: org.neo4j.helpers.PortBindException: Address localhost:31029 is already in use, cannot bind to it.
```

### Expected Behaviour

OK, in this case (with two test methods inside a single file) it would be possible to configure each test case with a new unused port.

But in general I expect shutdown of the embedded Neo4j instance together with `shutdown` block's invocation of `applicationContext?.stop()`.

### Environment Information

- **Operating System**: Ubuntu 18.04
- **Micronaut Version:** On your current branch `micronaut-2x` (but also seen locally with `2.0.0.M3`/`RC2`)
- **JDK Version:**
```
$ java -version
openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment (build 11.0.7+10-post-Ubuntu-2ubuntu218.04)
OpenJDK 64-Bit Server VM (build 11.0.7+10-post-Ubuntu-2ubuntu218.04, mixed mode, sharing)
```

I've set up a workspace like this (in a folder newly created):
1. `$ mn --version` emits `Micronaut Version: 2.0.0, JVM Version: 1.8.0_252`
2. `$ mn create-app --build gradle --features neo4j-bolt --inplace --lang java --test spock --jdk 11 foo`
3. Put the test case from above to a file like `src/test/groovy/Neo4jEmbeddedServerSpec.groovy`
4. `$ ./gradlew clean build`

Result:
```
Neo4jEmbeddedServerSpec > check embedded neo4j has been shut down by first test case FAILED
    org.spockframework.runtime.ConditionFailedWithExceptionError at Test.groovy:38
        Caused by: io.micronaut.context.exceptions.BeanInstantiationException at Test.groovy:38
            Caused by: io.micronaut.context.exceptions.ConfigurationException at Test.groovy:38
                Caused by: org.neo4j.server.ServerStartupException at Test.groovy:38
                    Caused by: org.neo4j.kernel.lifecycle.LifecycleException at Test.groovy:38
                        Caused by: java.lang.RuntimeException at Test.groovy:38
                            Caused by: org.neo4j.kernel.lifecycle.LifecycleException at Test.groovy:38
                                Caused by: org.neo4j.helpers.PortBindException
```

--- a/neo4j-bolt/src/main/java/io/micronaut/neo4j/bolt/embedded/EmbeddedNeo4jServer.java
+++ b/neo4j-bolt/src/main/java/io/micronaut/neo4j/bolt/embedded/EmbeddedNeo4jServer.java
@@ -29,6 +29,7 @@ import org.neo4j.kernel.configuration.BoltConnector;
 import org.neo4j.kernel.configuration.Connector;
 import org.neo4j.server.ServerStartupException;
 
+import javax.annotation.PreDestroy;
 import javax.inject.Singleton;
 import java.io.Closeable;
 import java.io.File;
@@ -254,6 +255,7 @@ public class EmbeddedNeo4jServer implements BeanCreatedEventListener<Neo4jBoltCo
         }
     }
 
+    @PreDestroy
     @Override
     public void close() throws IOException {
         serverControls.close();

--- a/neo4j-bolt/src/test/groovy/io/micronaut/neo4j/bolt/Neo4jEmbeddedServerSpec.groovy
+++ b/neo4j-bolt/src/test/groovy/io/micronaut/neo4j/bolt/Neo4jEmbeddedServerSpec.groovy
@@ -19,6 +19,7 @@ package io.micronaut.neo4j.bolt
 import org.neo4j.driver.Driver
 import io.micronaut.context.ApplicationContext
 import io.micronaut.core.io.socket.SocketUtils
+import spock.lang.Shared
 import spock.lang.Specification
 
 /**
@@ -27,11 +28,31 @@ import spock.lang.Specification
  */
 class Neo4jEmbeddedServerSpec extends Specification{
 
+    @Shared
+    int EMBEDDED_NEO4J_TCP_PORT = SocketUtils.findAvailableTcpPort()
+
     void "test neo4j embedded"() {
         given:
         ApplicationContext applicationContext = ApplicationContext.run(
-                'neo4j.uri':"bolt://localhost:${SocketUtils.findAvailableTcpPort()}",
+                'neo4j.uri':"bolt://localhost:${EMBEDDED_NEO4J_TCP_PORT}",
                 'neo4j.embedded.ephemeral':true
+        )
+
+        when:
+        Driver driver = applicationContext.getBean(Driver)
+
+        then:
+        driver.session().run('MATCH (n) RETURN n').size() == 0
+
+        cleanup:
+        applicationContext?.stop()
+    }
+
+    void "test neo4j embedded again on the same port"() {
+        given:
+        ApplicationContext applicationContext = ApplicationContext.run(
+            'neo4j.uri':"bolt://localhost:${EMBEDDED_NEO4J_TCP_PORT}",
+            'neo4j.embedded.ephemeral':true
         )
 
         when:

